planetary-social/nos
# Add participants in a thread to the mentions lookup

## Issues covered
#1568

## Description
This PR includes functionality to add authors of a thread to the mentions list . It contains the following changes:

- Adds a new function `mutualFolloweesCount` and `Array` extension function `sortByMutualFollowees` to the `Author` class file.
- Adds a new function `loadAuthorsFromReferences` to the `Event` class.
- Modifies the `NoteView`, `NoteComposer`, `NoteTextEditor` and `AuthorListView`.

## How to test
1. Build the code.
2. Look for a note with a lot of replies (I used Rabble's note to test) - This note (https://njump.me/note14cvuxagmang2mf9l2nfdu3h6aaeq7v68h4dsnm4t3av8kuf0xaese9hjw4) has some replies and inner replies.
3. Click on the reply button.
4. Tap the @ button or type "@" into the textfield.
5. Please check that:
    - Before you type in the search field, the authors in the thread of that note, and authors referenced in the note you are replying are in the mentions list.
    - After you type in the search field, the authors of the search result comes before the authors in the thread.
    - The list is sorted according to the number of mutual followees you have with the authors in the mentions list.

## Screenshots/Video
https://drive.google.com/file/d/1CDtuJENs52fWm7eAnV__SCV5RF97NM8K/view?usp=sharing
